### DISCUSSION TITLE
Improved Calendar Week View

### DISCUSSION THREAD

**Discussion by kitzler-walli** [kimai/kimai#4120] [General]
Is anybody interested in this improved calendar week view?
I added duration indication per item, per day and per week to the week view, and removed the half hour indicators on the left.
Let me know if anybody has interest in this "improvement" then I'll create a pull request.
![Untitled 2](https://github.com/kimai/kimai/assets/1926992/1f787618-e58e-41f7-b952-b2785c4256e1)

**Comment by kevinpapst** [comment]
Can you make the entry header configurable, same as this?
<img width="389" alt="Bildschirmfoto 2023-06-20 um 22 52 28" src="https://github.com/kimai/kimai/assets/533162/9069ce8c-9603-4eb1-9f76-366cd4d06908">

What do you mean by half hour indicator?

**Reply by kitzler-walli** [reply]
The content of the items is untouched - I just blurred it for my screenshot so that customer information won't be disclosed.
So the settings of the individual user or calendar settings are treated the same way as before.
For the item itself only the duration is added next to the start and endtime.

By default the slotLabelInterval is not set and that leads to labels like "14", "14:30", "15", ... which makes it harder to read.
By setting it to slotLabelInterval: "01:00" only the hour is displayed. 
To be honest this setting should be configurable separately but I didn't find the time to create a separate setting for it yet.

All the changes happened in the KaimaiCalendar.js widget and with fullcalendar 5 it was a lot harder to implement compared to version 3 as there are some events missing to alter the rendering and calculate overall sums.

**Reply by kevinpapst** [reply]
> By default the slotLabelInterval is not set and that leads to labels like "14", "14:30", "15", ... which makes it harder to read.
 
What slotInterval did you configure? I see only 30 minutes, if I increase to 00:15:00 slot interval (via settings).

<img width="109" alt="Bildschirmfoto 2023-06-21 um 08 38 41" src="https://github.com/kimai/kimai/assets/533162/b7f2c65d-98d0-4260-b313-f2454b821dac">

Anyway: I like the duration. Does it work on month view as well?

**Reply by kitzler-walli** [reply]
Maybe it's just me but the 6:30am, 7:30am, ... are irritating and unnecessary, therefore I removed them in my view.
If we could have a setting for slotLabelInterval everybody would be able to configure it the way he likes.

For the month view it would work as well, but currently doesn't.

**Reply by kevinpapst** [reply]
Too bad that there is no additional class besides `fc-timegrid-slot-label-cushion fc-scrollgrid-shrink-cushion` that is somehow identifiable. 
I assume your code uses some hooks to remove the slotLabelInterval div? If so: can it be used to add a custom css class `fc-kimai-timegrid-slot-label-auto`? Then we can hide the labels using custom css in a very generic fashion.

**Reply by kevinpapst** [reply]
@kitzler-walli  Can you share the code for the duration of each day?

**Reply by kitzler-walli** [reply]
I'll share it soon - too much to do currently

**Comment by kitzler-walli** [comment]
slotLabelInterval is a property of fullcalendar. According to the comments around the property you seem to have fiddled with it but you weren't happy with the results. Simply enabling a duration dropdown in the settings for it would be sufficient.

**Comment by pgassmann** [comment]
I am missing the sum of hours on top of each day in the calendar since upgrade to v2. These daily and weekly statistics are displayed in the screenshot by @kitzler-walli 
In my profile settings I enabled the "daily statistics in Timesheet" option. @kevinpapst what is that option for?

something else: In the calendar view, I miss the option to delete an entry. I can only add and update entries.

**Reply by pgassmann** [reply]
> something else: In the calendar view, I miss the option to delete an entry. I can only add and update entries.

I just noticed the right click feature.
![image](https://github.com/kimai/kimai/assets/460192/e1181d26-7515-4463-bf9d-fc8776db34a5)

A Delete button in the edit popup would be nice. Many users won't expect and discover the right click features.

**Reply by empx92** [reply]
Rightclick Function is nice, but it also cuts the usability on smartphones

**Reply by kevinpapst** [reply]
The calendar is one of the screens that is not made for mobile anyway.
But the context menu on mobile is always "long" click. 
That is common sense and has nothing to do with Kimai.

**Reply by empx92** [reply]
thank you for your information

**Comment by HenrivdGeest** [comment]
That screenshot of kitzler-walli looks exactly how we would love to see it! So far we really like the time entering via the calender, works great. The totals per slot/day/week are indeed very welcome.  Love to see that pull request!

**Comment by kitzler-walli** [comment]
I currently struggle to extract the changes as I was too lazy to create a branch first. I'll have to separate them from the rest to create a pull request. I'll get there soon.

**Comment by empx92** [comment]
Sadly only a JavaScript solution with some thing who could even be a bit better but it works for us:
<img width="612" alt="image" src="https://github.com/kimai/kimai/assets/106057957/8bb62ec2-dce5-44df-ba8a-3dea2a46ea9b">

**Comment by kitzler-walli** [comment]
Sorry for the delay but yesterday I created a pull request with the feature shown in my first post. https://github.com/kimai/kimai/pull/5427
